macros/cross_db_utils/listagg.sql

```diff
@@ -0,0 +1,50 @@
+{% macro listagg(measure, delimiter_text, order_by_clause='', limit_clause='') -%}
+  {{ return(adapter.dispatch('listagg', 'dbt_utils') (measure, delimiter_text, order_by_clause, limit_clause)) }}
+{%- endmacro %}
+
+
+{% macro default__listagg(measure, delimiter_text, order_by_clause, limit_clause) -%}
```

Comment:
Not all of your implementations support `limit_clause` - for the cross-db macros it's important that they all do the same thing, because package maintainers use them to get consistent behaviour across all the different adapters. 

There's a couple of options, depending on how committed you're feeling (either option is OK! This is why open source is great):
- Remove the limit clause altogether - if and when someone else comes along who wants to add support, they can do it separately. 
- Look into a way to get equivalent behaviour by using other patterns. For example this sort of thing: https://stackoverflow.com/questions/27405150/postgresql-string-agg-with-limited-number-of-elements (you could stick with the native code unless a limit clause is provided, and then switch to the alternative approach).

Reply:
alright! got this working (redshift was a beast) with the caveat that if there are instances of `delimiter_text` within your `measure`, you cannot include a `limit_num`